tests/unit_tests/simpll/DifferentialFunctionComparatorTest.cpp

```diff
@@ -314,6 +330,75 @@ class DifferentialFunctionComparatorTest : public ::testing::Test {
     }
 };
 
+/// Test fixture for testing DifferentialFunctionComparator
+/// for tests written in LLVM IR.
+class DFCLlvmIrTest : public DifferentialFunctionComparatorTest {
+  public:
+    std::string errorMsg = "";
+
+    DFCLlvmIrTest() {}
+
+    /// Tries to parse LLVM IR from string and prepares DFC for testing,
+    /// if parsing was unsuccessful returns false and the errorMsg is set.
+    bool prepare(const char *left, const char *right) {
+        ModL = stringToModule(left, CtxL);
+        if (ModL == nullptr) {
+            errorMsg = "Error while parsing left LLVM IR " + errorMsg;
+            return false;
+        }
+        ModR = stringToModule(right, CtxR);
+        if (ModR == nullptr) {
+            errorMsg = "Error while parsing right LLVM IR " + errorMsg;
+            return false;
+        }
+        FL = getFunByName(*ModL, "f");
+        if (FL == nullptr) {
+            errorMsg = "Error left LLVM IR: missing test function named `f`";
+            return false;
+        }
+        FR = getFunByName(*ModR, "f");
+        if (FR == nullptr) {
+            errorMsg = "Error right LLVM IR: missing test function named `f`";
+            return false;
+        }
+        prepareDFC();
+        return true;
+    }
+
+    /// Tries to parse string containing LLVM IR to module,
+    /// if unsuccessful sets error message.
+    std::unique_ptr<Module> stringToModule(const char *llvm, LLVMContext &ctx) {
+        SMDiagnostic err;
+
+        StringRef strRef(llvm);
+        std::unique_ptr<MemoryBuffer> memBuf =
+                MemoryBuffer::getMemBuffer(strRef);
+        MemoryBufferRef memBufRef(*memBuf);
+
+        std::unique_ptr<Module> mod(parseIR(memBufRef, err, ctx));
```

Comment:
I think that it should be possible to directly use `*memBuf` here:
```suggestion
        std::unique_ptr<Module> mod(parseIR(*memBuf, err, ctx));
```
and you can spare one variable.